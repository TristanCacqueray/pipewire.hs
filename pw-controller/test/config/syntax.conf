# a comment

connect input sink source # top level expr

when device:"a regex:.*"
  connect input sink source

# Nested when
when device:top-level
  # When can be nested
  when device:nested
    connect output source sink

  connect inout src sink

connect output r"a raw match" end
