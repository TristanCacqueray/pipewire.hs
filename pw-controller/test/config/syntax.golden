Right
    [ Connect
        { direction = Input
        , target = Target
            { kind = Port
            , matcher = r"sink"
            }
        , with = Target
            { kind = Port
            , matcher = r"source"
            }
        , redirect = Nothing
        }
    , When
        { condition = DeviceMatcher "^a regex:.*$"
        , body =
            [ Connect
                { direction = Input
                , target = Target
                    { kind = Port
                    , matcher = r"sink"
                    }
                , with = Target
                    { kind = Port
                    , matcher = r"source"
                    }
                , redirect = Nothing
                }
            ]
        }
    , When
        { condition = DeviceMatcher r"top-level"
        , body =
            [ When
                { condition = DeviceMatcher r"nested"
                , body =
                    [ Connect
                        { direction = Output
                        , target = Target
                            { kind = Port
                            , matcher = r"source"
                            }
                        , with = Target
                            { kind = Port
                            , matcher = r"sink"
                            }
                        , redirect = Nothing
                        }
                    ]
                }
            , Connect
                { direction = InOut
                , target = Target
                    { kind = Port
                    , matcher = r"src"
                    }
                , with = Target
                    { kind = Port
                    , matcher = r"sink"
                    }
                , redirect = Nothing
                }
            ]
        }
    , Connect
        { direction = Output
        , target = Target
            { kind = Port
            , matcher = r"a raw match"
            }
        , with = Target
            { kind = Port
            , matcher = r"end"
            }
        , redirect = Nothing
        }
    ]